File: code/web_dashboard/src/types/equipment.ts
// Mirrors backend app/schemas/equipment.py and app/models/enums.py exactly.

export type EquipmentStatus = 'AVAILABLE' | 'BOOKED' | 'ISSUED' | 'MAINTENANCE' | 'RETIRED';
export type EquipmentCondition = 'GOOD' | 'FAIR' | 'DAMAGED';

export interface Equipment {
  id: number;
  name: string;
  category: string;
  qr_code: string;
  status: EquipmentStatus;
  condition: EquipmentCondition;
  location: string | null;
  notes: string | null;
  added_at: string;
}

export interface EquipmentCreateRequest {
  name: string;
  category: string;
  qr_code: string;
  location?: string;
  notes?: string;
}

export interface EquipmentUpdateRequest {
  name?: string;
  category?: string;
  location?: string;
  notes?: string;
  status?: EquipmentStatus;
  condition?: EquipmentCondition;
}

File: code/backend/app/models/equipment.py
from datetime import datetime, timezone
from typing import Optional
from sqlmodel import Field, SQLModel
from app.models.enums import EquipmentCondition, EquipmentStatus


class Equipment(SQLModel, table=True):
    __tablename__ = "equipment"

    id: Optional[int] = Field(default=None, primary_key=True)
    name: str = Field(max_length=255)
    category: str = Field(max_length=100, index=True)
    qr_code: str = Field(max_length=255, unique=True, index=True)
    status: EquipmentStatus = Field(default=EquipmentStatus.AVAILABLE, index=True)
    condition: EquipmentCondition = Field(default=EquipmentCondition.GOOD)
    location: Optional[str] = Field(default=None, max_length=255)
    notes: Optional[str] = Field(default=None)
    added_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    added_by: int = Field(foreign_key="users.id")
